cg/services/file_delivery/fetch_file_service/fetch_analysis_files_service.py

```diff
@@ -0,0 +1,76 @@
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants import Workflow
+from cg.services.file_delivery.fetch_file_service.fetch_delivery_files_service import (
+    FetchDeliveryFilesService,
+)
+from housekeeper.store.models import File
+
+from cg.services.file_delivery.fetch_file_service.models import SampleFile, CaseFile, DeliveryFiles
+from cg.store.exc import EntryNotFoundError
+from cg.store.models import Case
+from cg.store.store import Store
+
+
+class FetchAnalysisDeliveryFilesService(FetchDeliveryFilesService):
+    """
+    Fetch the analysis files for a case from Housekeeper.
+    """
+
+    def __init__(self, status_db: Store, hk_api: HousekeeperAPI):
+        self.status_db = status_db
+        self.hk_api = hk_api
+
+    def get_files_to_deliver(self, case_id: str) -> DeliveryFiles:
+        """Return a list of analysis files to be delivered for a case."""
+        case: Case = self._get_case(case_id)
+        analysis_case_files: list[CaseFile] = self.get_analysis_case_delivery_files(case)
+        analysis_sample_files: list[SampleFile] = self._get_analysis_sample_delivery_files(
+            case=case
+        )
+        return DeliveryFiles(case_files=analysis_case_files, sample_files=analysis_sample_files)
+
+    def _get_case(self, case_id: str):
+        """Get the case."""
+        if case := self.status_db.get_case_by_internal_id(internal_id=case_id):
+            return case
+        raise EntryNotFoundError(f"Case {case_id} not found.")
```

Comment:
Doesn't feel right to have this general method here, but I guess it's ok for now. Seems like this checks will end up in a service of its own in the future

Reply:
Yes, this is a weird one. Ideally this check is part of the CRUD. However, this would require a refactoring of all the places to handle the error properly. I don't like having this here.

Reply:
It might be better to accept the current design where the return can be none instead of spreading helper methods like this, your call